### PR TITLE
[Tests-Only] Assert created user

### LINKED ISSUE #37828: theseUsersHaveBeenCreatedUsingTheOccCommand is missing a check
https://github.com/owncloud/core/blob/master/tests/acceptance/features/bootstrap/OccUsersGroupsContext.php#L129

`theseUsersHaveBeenCreatedUsingTheOccCommand()` ends with:

```
\in_array($tableUsername, $usernameArray, true);
```

which does nothing - the true/false value is not used.

Fix it so that it actually asserts something useful.

### PR DESCRIPTION
**Pull request by swoichha**
## Description
Assert created users with the table of users required to be created.

## Related Issue
- Fixes https://github.com/owncloud/core/issues/37828#

## How Has This Been Tested?
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/OccUsersGroupsContext.php [hunk 1]
```diff
@@ -130,13 +130,13 @@ public function theseUsersHaveBeenCreatedUsingTheOccCommand(TableNode $table) {
...
 		$createdUsersList = $this->featureContext->getCreatedUsers();
-		$tableUsername = [];
-		$displayName = [];
 		$usernameArray = \array_keys($createdUsersList);
 		foreach ($table as $row) {
-			\array_push($tableUsername, $row['username']);
+			Assert::assertTrue(
+				\in_array($row['username'], $usernameArray),
+				"User: " . $row['username'] . " was not created"
+			);
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/OccUsersGroupsContext.php:138]: Ok, I think there is something weird going on here. We want to check that the user we just created is actually in the server, but we only check it against the `$username` array which is just a local array stored in the test-runner, so the user is obviously gonna be there even if the user creation failed.
I think you would want to get the users list from the server and check there.

Comment (phil-davis) [reply tests/acceptance/features/bootstrap/OccUsersGroupsContext.php:138]: The code here was already strange. 

You could just loop through each user and assert that `featureContext->userShouldExist` is `true`

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

**Review by phil-davis — APPROVED** [review]

**Pull request merged**